tools/signing/sign_plain_key.go

```diff
@@ -0,0 +1,61 @@
+package signing
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	// NOTE MUST BE > 1.14 for this fix
+	// https://github.com/ethereum/go-ethereum/pull/28945
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/mcms_proposal"
+)
+
+// Just run this locally to sign from the ledger.
+func signPlainKey(privateKeyHex string) {
+	// Load file
+	proposal, _ := ProposalFromFile(os.Args[0])
+	err := proposal.Validate()
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	executor, err := proposal.ToExecutor(make(map[string]mcms_proposal.ContractDeployBackend)) // TODO: pass in a real backend
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Get the signing hash
+	payload, err := executor.SigningHash()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Load private key
+	privateKey, err := crypto.HexToECDSA(privateKeyHex)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Sign the payload
+	sig, err := crypto.Sign(payload.Bytes(), privateKey)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Sign the payload
+	unmarshalledSig := mcms_proposal.Signature{}
+	err = json.Unmarshal(sig, &unmarshalledSig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Add signature to proposal
```

Comment:
nit: could insert it sorted

Reply:
don't really see a significant advantage to doing this, seems liike more code for what would a not noticeable difference in performance at the scale we expect this array to be at

Reply:
The value isn't for performance, but for deterministic ordering when we process the proposal afterwards. We either insert the signatures sorted, or we sort the signatures later when executing / checking roots. 

Just a tiny detail, but it feels cleaner to me

Reply:
> but for deterministic ordering when we process the proposal afterwards. We either insert the signatures sorted, or we sort the signatures later when executing / checking roots.

This is only true if we can guarantee that signatures are only ever inserted in this way, which we can't always guarantee because after signing, library users can theoretically reorder the sigs array. IMO, We should only transform when we need to transform to ensure that we can guarantee that when we try to SetRoot we don't get any failures

Reply:
This is also related to another comment above but I'll reiterate here: 

>  because I'm not sure if this function will be used elsewhere other than SetRoot. If its only used in SetRoot, it doesn't matter, but if it is gonna be used elsewhere, I wanna avoid adding in the complexity of sorting in places where its not relevant to the library consumers. sorting is only relevant when broadcasting onchain so I think we should confine that logic to that function